Remove ErrorEventInit's error default

cdumez opened the pull request:
#### c649008d0fd5c0f083c8a13e867b5f8042f2ecde
<pre>
Remove ErrorEventInit&apos;s error default
<a href="https://bugs.webkit.org/show_bug.cgi?id=241259">https://bugs.webkit.org/show_bug.cgi?id=241259</a>
&lt;rdar://94793559&gt;

Reviewed by Darin Adler.

Update ErrorEventInit&apos;s error default to be undefined instead of null, as per:
- <a href="https://github.com/whatwg/html/pull/7983">https://github.com/whatwg/html/pull/7983</a>

* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/event-handler-processing-algorithm-error/document-synthetic-errorevent-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/event-handler-processing-algorithm-error/document-synthetic-errorevent.html:
* Source/WebCore/dom/ErrorEvent.cpp:
(WebCore::ErrorEvent::error):
* Source/WebCore/dom/ErrorEvent.idl:

Canonical link: <a href="https://commits.webkit.org/251797@main">https://commits.webkit.org/251797@main</a>
</pre>

darinadler commented on Source/WebCore/dom/ErrorEvent.cpp:
```diff
@@ -49,7 +49,7 @@ ErrorEvent::ErrorEvent(const AtomString& type, const Init& initializer, IsTruste
     , m_lineNumber(initializer.lineno)
     , m_columnNumber(initializer.colno)
-    , m_error(initializer.error)
+    , m_error(initializer.error ? initializer.error : jsUndefined())
```
I have no doubt that this is necessary, but it doesn’t seem quite right. Seems to me that an `any` without an initial value should be initialized to `jsUndefined()` by the bindings. It doesn’t seem like it should be the constructor’s job to check for the "TDZ" case, which I guess is what JavaScript experts call a C++ null value for `JSValue`.

cdumez replied:
I think an alternative that might be clearer would be:
```
    struct Init : EventInit {
        String message;
        String filename;
        unsigned lineno { 0 };
        unsigned colno { 0 };
        JSC::JSValue error { JSC::jsUndefined() };
    };
```

Initialize error to jsUndefined() by default in ErrorEvent::Init.

cdumez committed:
Remove ErrorEventInit's error default
https://bugs.webkit.org/show_bug.cgi?id=241259
<rdar://94793559>

Reviewed by Darin Adler.

Update ErrorEventInit's error default to be undefined instead of null, as per:
- https://github.com/whatwg/html/pull/7983

* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/event-handler-processing-algorithm-error/document-synthetic-errorevent-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/event-handler-processing-algorithm-error/document-synthetic-errorevent.html:
* Source/WebCore/dom/ErrorEvent.cpp:
(WebCore::ErrorEvent::error):
* Source/WebCore/dom/ErrorEvent.idl:

Canonical link: https://commits.webkit.org/251797@main
